classes/note/Note.php

```
@@ -3,176 +3,148 @@
      */
-    public function setTitle($title)
+    public function scopeWithUserId(Builder $query, int $userId): Builder
     {
-        $this->setData('title', $title);
+        return $query->where('userId', $userId);
     }
 
     /**
-     * get note type
-     *
-     * @return int
+     * Scope a query to only include notes with a specific assoc type and assoc ID.
      */
-    public function getAssocType()
+    public function scopeWithAssoc(Builder $query, int $assocType, int $assocId): Builder
     {
-        return $this->getData('assocType');
+        return $query->where('assoc_type', $assocType)
+                     ->where('assoc_id', $assocId);
     }
 
     /**
-     * set note type
-     *
-     * @param int $assocType
+     * Scope a query to only include notes with a specific type.
      */
-    public function setAssocType($assocType)
+    public function scopeWithType(Builder $query, int $type): Builder
     {
-        $this->setData('assocType', $assocType);
+        return $query->where('type', $type);
     }
 
     /**
-     * get note assoc id
-     *
-     * @return int
+     * Scope a query to only include notes with a specific type.
      */
-    public function getAssocId()
+    public function scopeWithContents(Builder $query, string $contents): Builder
     {
-        return $this->getData('assocId');
+        return $query->where('contents', $contents);
     }
 
     /**
-     * set note assoc id
-     *
-     * @param int $assocId
+     * Scope a query to a specific sort order.
      */
-    public function setAssocId($assocId)
+    public function scopeWithSort(Builder $query, int $orderBy = NOTE_ORDER_DATE_CREATED, int $sortDirection = DAO::SORT_DIRECTION_DESC): Builder
     {
-        $this->setData('assocId', $assocId);
+        // Sanitize sort ordering
+        $orderSanitized = match ($orderBy) {
+            self::NOTE_ORDER_ID => 'note_id',
+            default => 'date_created',
```

```suggestion
            self::NOTE_ORDER_DATE_CREATED => 'date_created',
```
This way unsupported incoming parameters will result in an exception rather than silently ignoring it.